### PR TITLE
Fixes #7174: Allow rudder-dev base on a local branch

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7174

### CODE REVIEW

File: scripts/rudder-dev/rudder-dev [hunk 1]
```diff
@@ -766,8 +768,11 @@ def create_branch(ticket, base=None):
 
   # manage the original branch
   if base is not None:
-    # fetch base branch
-    branch_name = fetch_branch_from_ticket(base)
+    if re.match(r'^.\d*$', base):
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev]: Do you think it should be \d+ ?

File: scripts/rudder-dev/rudder-dev [hunk 2]
```diff
@@ -902,7 +907,11 @@ def rebase(comment=None, base=None):
 
   # fetch before rebasing
   if base is not None:
-    branch_name = fetch_branch_from_ticket(base)
+    if re.match(r'^.\d*$', base):
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev]: Same question here

**Comment by peckpeck** [comment]
PR updated

File: scripts/rudder-dev/rudder-dev (hunk 1 above)
Comment (peckpeck) [inline scripts/rudder-dev/rudder-dev]: No, the first character can be #,i or \d, and if it is already a \d it can be a single digit ticket id.

Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev]: I think all signle digit ticket id are fixed ;) but you are right :)

**Pull request merged**